### DISCUSSION TITLE
Let us select light shape for any light

### DISCUSSION THREAD

**Discussion by BlackMoons** [FakeFishGames/Barotrauma#13021] [Ideas and suggestions]
Can we please select what light texture is used for a light? Ie give any light a cone, point or double flood? 

Ie, just a little dropdown box that lets you pick from the Barotrauma\Content\Lights files? (Or a new 'light texture' object that is just a light texture filename, human readable name, and an origin point, so mods could add their own light shapes for sub builders/mappers to use?)

Also, a half flood would be amazing (light_fluorescent_L2.png except only half) for 'on the hull of ship' lights without needing shadow casting to prevent them bleeding inside.

**Comment by BlackMoons** [comment]
![light_fluorescent_Half](https://github.com/Regalis11/Barotrauma/assets/15382398/91d8d2bc-0eb7-41f3-8ec8-1a9d9827d2cf)

Here is an example of 'half flood' (Unidirectional flood?) 

Here is a ship outside lit with the normal double flood with shadow casting on:

![image](https://github.com/Regalis11/Barotrauma/assets/15382398/3172edb8-952e-4c0e-9def-dbca2fd984fd)

And here is it with 'draw behind sub', Not exactly desirable as it makes the entire hull pitch black:

![image](https://github.com/Regalis11/Barotrauma/assets/15382398/79794c1a-3040-4ec2-acbc-1dee2b34fe15)

And here is it lit entirely by the 'half flood' texture above, with cast shadows off:

![image](https://github.com/Regalis11/Barotrauma/assets/15382398/256fbb74-5827-40fb-a5ae-5971f9a8c262)

As you can see, in most cases it looks exactly like the double flood with shadows on. Given that exterior lights generally have the range cranked up, I feel its worthwhile if just from a performance standpoint to have a unidirectional flood light to avoid having to use shadow casting so much. In several cases I feel it looks better then the double flood even (like around the nose is more consistent with the rest of the ship now)

Plus it could be used indoors to light a room from the ceiling without worrying about bleeding too much into the room above it, giving a more moody look as it leaves the upper corners dark.

**Comment by ubertpendragon** [comment]
So true bestie

https://github.com/Regalis11/Barotrauma/discussions/9810

**Reply by BlackMoons** [reply]
So, extensively commented and upvoted discussion from a year ago. 

Easy to implement at the most basic level, like just letting you select light map texture (aka any texture file in the content\lights dir*) for existing lights, add like 5 new unidirectional light options, and have that rotate when you rotate the light like existing lightmaps do would be huge in reducing how many shadow casting lights most ships need. Multiple people independently raising discussions asking for this feature... 

Can we please see some progress on this feature developers? Or can maybe someone code it up and make a pull request? 


(* a new lighttexture XML object with an origin, user readable name (presumably with translation support) and addable by mods would be very cool too. Then people could easily add their own lightmaps without having to duplicate every light that they want to use them)
